### PR TITLE
feat(ingest/snowflake): optionally emit all upstreams irrespective of recipe pattern

### PR DESCRIPTION
**Pull request by mayurinehate**
Since the upstreams are not minted automatically, this change does not create ghost entities. Also, UI currently hides the non-minted upstreams.
If separate recipes are used to ingest from different snowflake databases in same snowflake account, one can set the below config to emit all upstreams :

`validate_upstreams_against_patterns: false`

However **Having single recipe for a snowflake account remains the first recommended solution.**

Also added tests for snowflake legacy lineage (default lineage method as of now.)
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_legacy.py [hunk 1]
```diff
@@ -228,36 +228,32 @@ def _populate_table_lineage(self):
     def get_table_upstream_workunits(self, discovered_tables):
         if self.config.include_table_lineage:
             for dataset_name in discovered_tables:
-                if self._is_dataset_pattern_allowed(
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_legacy.py:231]: This pattern check is not required, as its already present on discovered_tables [here](https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py#L529).

File: metadata-ingestion/tests/integration/snowflake/snowflake_privatelink_golden.json [hunk 2]
```diff
@@ -869,12 +2527,30 @@
...
 {
     "entityType": "dataset",
-    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,test_db.test_schema.table_9,PROD)",
+    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,test_db.test_schema.table_1,PROD)",
     "changeType": "UPSERT",
     "aspectName": "upstreamLineage",
     "aspect": {
-        "value": "{\"upstreams\": [{\"auditStamp\": {\"time\": 0, \"actor\": \"urn:li:corpuser:unknown\"}, \"dataset\": \"urn:li:dataset:(urn:li:dataPlatform:snowflake,test_db.test_schema.table_2,PROD)\", \"type\": \"TRANSFORMED\"}]}",
-        "contentType": "application/json"
+        "json": {
```
Comment (mayurinehate) [inline metadata-ingestion/tests/integration/snowflake/snowflake_privatelink_golden.json:2534]: This is the main change where `test_db.test_schema.table_1` now has `other_db.other_schema.table_1` as upstream along with its original upstream `test_db.test_schema.table_2`.

All other changes are due to 
1. Convert serialised MCPs to canonical form for better readability
2. Ordering changes - all lineage mcps are emitted at the end after schema ingestion is over.

**Review by asikowitz — APPROVED** [review]
Do we want to support customers making multiple ingestion pipelines for the same source? I thought our solution here was to combine into one. However, if this is something we want to support, then this looks good to me

**Comment by mayurinehate** [comment]
> Do we want to support customers making multiple ingestion pipelines for the same source? I thought our solution here was to combine into one. However, if this is something we want to support, then this looks good to me

The solution of combining recipes into one is the recommended solution, hence this config is disabled by default. The config option is only if for some reason, recipes per database need to be kept separate.

**Pull request merged**